community/values/src/main/java/org/neo4j/values/virtual/ListValue.java

```diff
@@ -315,14 +170,20 @@ public AnyValue value( int offset )
         @Override
         public AnyValue[] asArray()
         {
-            return values.toArray( new AnyValue[values.size()] );
+            return values.toArray( new AnyValue[0] );
```

Comment:
How can you be sure about that 0?

Reply:
The size of the array will be handled by `toArray`. Back in the day using a presized array was actually always recommended but in later versions of java it is actually faster to use an empty array because it will use intrinsics. If you time over, https://shipilev.net/blog/2016/arrays-wisdom-ancients/.

Intellij actually warns you about using presized arrays nowadays, which is why the reason I noticed this one.